### PR TITLE
Rewrite the 'PivotGrid - Drill Down' demo description

### PR DESCRIPTION
**Pull request by AlexJameson**

### CODE REVIEW

File: JSDemos/Demos/PivotGrid/DrillDown/description.md [hunk 1]
```diff
@@ -1 +1,5 @@
-Drill-down support within the PivotGrid component allows you to&nbsp;retrieve the list of&nbsp;records used to&nbsp;calculate a&nbsp;specific summary value. To&nbsp;see drill downs in&nbsp;action, click a&nbsp;summary cell and view the records that formed the basis of&nbsp;the summary value.
\ No newline at end of file
+PivotGrid supports the drill down operation that allows users to retrieve the list of facts used to calculate a specific summary value.
```
Comment (pavel-2020) [inline JSDemos/Demos/PivotGrid/DrillDown/description.md]: I presume users want to _display_ the list of facts. Also, "fact" is a term that may be unfamiliar for novices. Thus the suggestion is
```suggestion
PivotGrid supports the drill down operation that allows users to display the list of [facts](/Documentation/Guide/UI_Components/PivotGrid/Visual_Elements/#Summary_Values) (records) used to calculate a specific summary value. 
```

File: JSDemos/Demos/PivotGrid/DrillDown/description.md [hunk 2]
```diff
@@ -1 +1,5 @@
-Drill-down support within the PivotGrid component allows you to&nbsp;retrieve the list of&nbsp;records used to&nbsp;calculate a&nbsp;specific summary value. To&nbsp;see drill downs in&nbsp;action, click a&nbsp;summary cell and view the records that formed the basis of&nbsp;the summary value.
\ No newline at end of file
+PivotGrid supports the drill down operation that allows users to retrieve the list of facts used to calculate a specific summary value. 
+
+Use the [createDrillDownDataSource(options)](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Methods/#createDrillDownDataSourceoptions) method to access underlying facts for a summary value. This method returns a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) instance that contains a list of facts that form the basis of the summary value.
```
Comment (pavel-2020) [inline JSDemos/Demos/PivotGrid/DrillDown/description.md]: > access underlying facts for a summary value

is similar to
> contains a list of facts that form the basis of the summary value
```suggestion
Use the [createDrillDownDataSource(options)](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Methods/#createDrillDownDataSourceoptions) method to return a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) instance that contains a list of facts for the summary value.
```

File: JSDemos/Demos/PivotGrid/DrillDown/description.md [hunk 3]
```diff
@@ -1 +1,5 @@
...
+
+This demo shows how to display a list of facts in a [Popup](/Documentation/ApiReference/UI_Components/dxPopup/) that contains a [DataGrid](/Documentation/ApiReference/UI_Components/dxDataGrid). We use the [onCellClick](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#onCellClick) handler to retrieve a cell's data and call the **createDrillDownDataSource(options)** within this handler. Then we pass the created **DataSource** to the **DataGrid**'s [dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#dataSource) property. As a result, you can click a pivot grid cell to display a popup. This popup shows a list of facts used to calculate the summary value from the clicked cell.
```
Comment (pavel-2020) [inline JSDemos/Demos/PivotGrid/DrillDown/description.md]: Are there recommended components where to display a drill-down?

File: JSDemos/Demos/PivotGrid/DrillDown/description.md (hunk 1 above)
Comment (AlexJameson) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md]: The thing is that this method only returns collected data but does not display it anywhere. It is up to user to decide how to use it later.

About facts - I tend to agree.

File: JSDemos/Demos/PivotGrid/DrillDown/description.md (hunk 2 above)
Comment (AlexJameson) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md]:
```suggestion
Use the [createDrillDownDataSource(options)](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Methods/#createDrillDownDataSourceoptions) method to return a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) instance that contains a list of facts for a summary value.
```

File: JSDemos/Demos/PivotGrid/DrillDown/description.md (hunk 3 above)
Comment (AlexJameson) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md]: No, I don't think so. There are too many scenarios how users can process this data but I'll ask the team about it.

Comment (pavel-2020) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md]: If there were such components, would it be good to list them in the docs?

---

Shortened the para:
```suggestion
In this demo, we use the [onCellClick](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#onCellClick) handler to create a [DataGrid](/Documentation/ApiReference/UI_Components/dxDataGrid) of facts and display it inside a [Popup](/Documentation/ApiReference/UI_Components/dxPopup/). To do this, we pass the clicked cell's data as an argument for **createDrillDownDataSource** and bind the returned [dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#dataSource) to the DataGrid.
```

File: JSDemos/Demos/PivotGrid/DrillDown/description.md (hunk 1 above)
Comment (RomanTsukanov) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md]: For lack of any guideline within DX regarding the term "drill down", let's be guided by the [Microsoft recommendations](https://docs.microsoft.com/ru-ru/style-guide/a-z-word-list-term-collections/d/drill-down-drill-up-drill-through).

> Also, "fact" is a term that may be unfamiliar for novices.

Agreed, but I think the adding "records" in brackets would be enough to clear things up.

```suggestion
PivotGrid supports the drill-down operation that allows users to analyze individual facts (records) used to calculate a specific summary value.
```

File: JSDemos/Demos/PivotGrid/DrillDown/description.md [hunk 4]
```diff
@@ -1 +1,5 @@
-Drill-down support within the PivotGrid component allows you to&nbsp;retrieve the list of&nbsp;records used to&nbsp;calculate a&nbsp;specific summary value. To&nbsp;see drill downs in&nbsp;action, click a&nbsp;summary cell and view the records that formed the basis of&nbsp;the summary value.
\ No newline at end of file
+PivotGrid supports the drill down operation that allows users to retrieve the list of facts used to calculate a specific summary value. 
+
+Use the [createDrillDownDataSource(options)](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Methods/#createDrillDownDataSourceoptions) method to return a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) instance that contains a list of facts for a summary value.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/PivotGrid/DrillDown/description.md]: > Use the [createDrillDownDataSource(options)

Methods are called. Avoiding overused words is always a good thing.

> to return a [DataSource

It's methods that "return" something. We address developers, and they want to "get" or "obtain" the list of facts.

```suggestion
Call the [createDrillDownDataSource(options)](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Methods/#createDrillDownDataSourceoptions) method to get a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) instance that contains a list of facts for a summary value.
```

File: JSDemos/Demos/PivotGrid/DrillDown/description.md (hunk 3 above)
Comment (RomanTsukanov) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md]: > This demo shows how to [do X]

Do not use this intro for custom scenarios. A developer may have a different scenario. Use "This demo does X" or "In this demo, X is done".

> We use the [onCellClick](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#onCellClick) handler to retrieve a cell's data

You haven't told yet that a cell should be clicked. You tell it a couple of sentence after: "As a result, you can click a pivot grid cell to display a popup.".

> Then we pass the created **DataSource** to the **DataGrid**'s [dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#dataSource) property.

This is about the actual scenario. I think we can remove sentence.

```suggestion
This demo displays a list of facts within a [DataGrid](/Documentation/ApiReference/UI_Components/dxDataGrid) in a [Popup](/Documentation/ApiReference/UI_Components/dxPopup/) window. To open the window, click a pivot grid cell. We use the [onCellClick](/Documentation/ApiReference/UI_Components/dxPivotGrid/Configuration/#onCellClick) event handler to retrieve the cell's data and pass it as an argument to the **createDrillDownDataSource(options)** method.
```

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/PivotGrid/DrillDown/description.md (hunk 1 above)
Comment (alexey-zolotarev) [inline JSDemos/Demos/PivotGrid/DrillDown/description.md]: This and two following sentences all have the same ending (retrieve the list of facts used to calculate a specific summary value; access underlying facts for a summary value; a list of facts that form the basis of the summary value.). Can we simplify or shorten this text ?

File: JSDemos/Demos/PivotGrid/DrillDown/description.md [hunk 5]
```diff
@@ -1 +1,5 @@
-Drill-down support within the PivotGrid component allows you to&nbsp;retrieve the list of&nbsp;records used to&nbsp;calculate a&nbsp;specific summary value. To&nbsp;see drill downs in&nbsp;action, click a&nbsp;summary cell and view the records that formed the basis of&nbsp;the summary value.
\ No newline at end of file
+PivotGrid supports the drill-down operation that allows you to retrieve individual facts (records) used to calculate a specific summary value.
```
Comment (albertov05) [inline JSDemos/Demos/PivotGrid/DrillDown/description.md:1]: What do you mean by "facts"? Is there a more specific term we can use?
If possible, can I try out this demo?

Comment (AlexJameson) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md:1]: Hello. This is a basic term that is familiar to people who work with data (analysts, data engineers, etc). You can find it in almost any related article: https://doc.sitecore.com/en/developers/90/sitecore-experience-platform/facts-and-dimensions.html

Basically it is just a single value (record).

Comment (albertov05) [reply JSDemos/Demos/PivotGrid/DrillDown/description.md:1]: Okay, thanks for the clarification. :)

**Pull request merged**